File: base/basic.py
arr = [1,1,0,1]
place = 1 
num = 0 

for i in range(len(arr)): 
    num += place*arr[i]
    place *= 10
print(num)
real_num = 0 
pl = 1
while num >0: 
    r = num%10 
    real_num += r*pl 
    pl*=10 
    num//=10 
print(real_num)

File: pointers/pointers and array/basic.cpp
#include<iostream>
using namespace std;

int main()
{
    // int num = 6;
    // int *ptr = &num;

    // cout<<"addres of num : "<<&num;
    // cout<<"\naddres of num : "<<++ptr;
    // cout<<"\nvalue of num : "<<*ptr;
    // cout<<"\nvalue of num : "<<num;

    int *arr = new int[4];
    *(arr + 0) = 1;
    *(arr + 1) = 2;
    *(arr + 2) = 3;

    for(int i=0; i<4; i++)
    {
        cout<<*(arr + i)<<endl;
        cout<<(arr + i)<<endl;
    }
    cout<<(arr + 4);
    delete []arr;
}